integration/tests/services/test_ws_subscriber.py

```diff
@@ -0,0 +1,223 @@
+import json
+import random
+import string
+from types import SimpleNamespace
+
+import allure
+import pytest
+import websockets
+from eth_utils import keccak
+
+from integration.tests.basic.helpers.rpc_checks import assert_fields_are_hex
+from integration.tests.basic.test_rpc_calls import TestRpcCalls
+from integration.tests.services.helpers.basic import ws_receive_all_messages
+
+TEST_STAND = "ws://159.69.194.181:8282"
+
+
+class ETH:
+    def __init__(self, params: list, method: str):
+        self.jsonrpc = 2.0
+        self.id = 1
+        self.method = method
+        self.params = params
+
+
+class Subscribe(ETH):
+    def __init__(self, params: list):
+        super().__init__(params, "eth_subscribe")
+
+
+class Unsubscribe(ETH):
+    def __init__(self, subscription_id: str):
+        super().__init__([subscription_id], "eth_unsubscribe")
+
+
+@allure.feature("Websocket Subscriber")
+@allure.story("Subscribe to events")
+class TestSubscriber(TestRpcCalls):
+
+    async def test_eth_subscribe(self):
+        async with websockets.connect(TEST_STAND) as ws:
+            data = Subscribe(["newHeads"])
+            await ws.send(json.dumps(data.__dict__))
+            r = await ws.recv()
+            response = json.loads(r, object_hook=lambda d: SimpleNamespace(**d))
+            subscription = response.result
+            assert len(subscription) > 0
+
+            self.send_neon(self.sender_account, self.recipient_account, 0.1)
+
+            r = await ws.recv()
+            response = json.loads(r, object_hook=lambda d: SimpleNamespace(**d))
+            assert response.params.subscription == subscription
+            assert len(vars(response.params.result).items()) > 0
+
+            while len(ws.messages) > 0:  # clear all received messages
+                ws.messages.popleft()
+
+            data = Unsubscribe(subscription)
+            await ws.send(json.dumps(data.__dict__))
+            r = await ws.recv()
+            response = json.loads(r, object_hook=lambda d: SimpleNamespace(**d))
+            assert response.result == 'true'
+
+    @pytest.mark.parametrize("param_fields", [("address", "topics"), ("topics",), ("address",), ()], ids=str)
+    async def test_logs(self, param_fields):
+        async with websockets.connect(TEST_STAND) as ws:
+            event_caller, _ = self.web3_client.deploy_and_get_contract(
```

Comment:
you can deploy contract not inside async with, before connect to websockets and will be better if you deploy it once before this test and use in all parametrized tests

Reply:
Good idea, added fixture.